components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -1994,8 +1997,14 @@ void JsonRpcService::OnGetBlockByNumber(GetBlockByNumberCallback callback,
 }
 
 /*static*/
-bool JsonRpcService::IsValidDomain(const std::string& domain) {
-  static const base::NoDestructor<re2::RE2> kDomainRegex(kDomainPattern);
+bool JsonRpcService::IsValidEnsDomain(const std::string& domain) {
+  static const base::NoDestructor<re2::RE2> kDomainRegex(kEnsDomainPattern);
+  return re2::RE2::FullMatch(domain, *kDomainRegex);
+}
+
+/*static*/
+bool JsonRpcService::IsValidSnsDomain(const std::string& domain) {
+  static const base::NoDestructor<re2::RE2> kDomainRegex(kSnsDomainPattern);
```

Comment:
Should we lowercase this domain before checking the kDomainRegex? I'm not finding documentation of SNS supported characterset, but this seems a bit overly restrictive.

Reply:
![Screenshot 2024-01-22 at 4 29 46 PM](https://github.com/brave/brave-core/assets/23125059/3999346a-d82e-426e-8fd5-18b6e17537c4)